Display average of score field for all reviews

gmurtaza00 opened the pull request:
Adds scores and an average score to the review outcomes in submissions tables.

![image](https://user-images.githubusercontent.com/111306331/228357674-f03705f8-ffaa-49a4-973a-1064456a0ffe.png)

Open to suggestion on this one, when no review has been submitted. We can either remove Avg. score or add a placeholder for 0 average

![image](https://user-images.githubusercontent.com/111306331/228357823-1d3cf745-98b7-407c-8305-30d1a5c56697.png)

Closes #3304

chriszs committed:
feat: display average of scores for  reviews

chriszs committed:
fix: remove hover state, bug for review outcomes

gmurtaza00 committed:
Average Total Score visible in the Reviews & assignees box

gmurtaza00 committed:
Filter improved to exclude the reviewers who have not reviewed yet

chriszs committed:
fix: change average review score computation

frjo commented on hypha/static_src/src/sass/apply/custom/_custom.scss:
```diff
@@ -1 +1,7 @@
+
+.list__item--average_score {
+    color: $color--mid-dark-grey;
+    font-weight: 400;
```
Please use variables for font weight as well.

gmurtaza00 committed:
Sorted imports

frjo reviewed (changes requested):
Looks good, just one very minor fix before I send this to testing.

gmurtaza00 committed:
use of variables in css instead of hard coded value

theskumar commented:
Thoughts on moving this here, @frjo @gmurtaza00 

![Screenshot 2023-04-04 at 07 45 07@2x](https://user-images.githubusercontent.com/236356/229723576-d7c6efbd-84e9-453c-b9e4-b97384684d67.jpg)

gmurtaza00 commented:
> Thoughts on moving this here, @frjo @gmurtaza00
> 
> ![Screenshot 2023-04-04 at 07 45 07@2x](https://user-images.githubusercontent.com/236356/229723576-d7c6efbd-84e9-453c-b9e4-b97384684d67.jpg)

![image](https://user-images.githubusercontent.com/111306331/229860918-55b93076-6bf2-419c-84b2-b26da91f88c7.png)
![image](https://user-images.githubusercontent.com/111306331/229860850-3516e46e-1406-41ef-a968-996a31a7d6f4.png)

Does this seem alright? @theskumar @frjo

theskumar commented:
That what's I was thinking, the red color was more of an indicator from my annotation tool. You can keep the font color same as the body font `color` (black) and `font-weight` and rest looks good to me.  @gmurtaza00 

![Screenshot 2023-04-04 at 18 54 08@2x](https://user-images.githubusercontent.com/236356/229892164-35de9aef-36e9-4382-9d6a-9b488b43d25e.jpg)

gmurtaza00 committed:
Changed the placement and styling of average score.

frjo commented on hypha/static_src/src/sass/apply/custom/_custom.scss:
```diff
@@ -1 +1,10 @@
+    font-size: map-get($font-sizes, zeta);
+    display: inline;
+    padding-left: 2px;
+}
```
`_custom.scss` should be empty in stock hypha. It is used to add your custom css without worrying about merge conflicts.

frjo commented on hypha/static_src/src/sass/apply/components/_list.scss:
```diff
@@ -53,19 +53,6 @@
-                    text-align: right;
-                }
-            }
-
```
Good, this is some old left over css. Thanks for noticing it.

frjo reviewed (changes requested):
Move the css from custom file and we are good I think.

gmurtaza00 committed:
moved css from _custon.scss to _list.scss

gmurtaza00 committed:
Removed 'NA' from scores array when computing average of review scores.
* When a review form has no score field, 'NA' is stored in score variable and that is equivalent to 99 according to options.py file.
* This was causing issue while computing average, so removed any score that is 'NA' from score array.

gmurtaza00 committed:
Changed position of Avg. Score.
Placed it on the top of review and assignee box next to the traffic light on recommendation of Saurabh kumar and Fredrik Jonsson.

gmurtaza00 committed:
Removed unnecessary scss

gmurtaza00 committed:
Removed average score subquery. Used custom filter for calculating average review score and displayed individual score using get_score_display method.

gmurtaza00 commented:
Initially NA score was being display as 99 and average was calculated using this score. Rebasing resolved this issue. Now score is displayed correctly and average is also being calculated correctly.

![image](https://user-images.githubusercontent.com/111306331/234944311-8ded5792-d502-47d5-b134-e3ba8ab73ad4.png)

frjo commented:
Closing this since this view is being replaced.
